bittide/src/Data/Constraint/Nat/Extra.hs

```diff
@@ -40,6 +44,20 @@ instance KnownNat n => KnownNat1 $(nameToSymbol ''OneMore) n where
     _ -> SNatKn 1
   {-# INLINE natSing1 #-}
 
+type family SubZero (a :: Nat) (b :: Nat) :: Nat where
+  SubZero 0 b = 0
```

Comment:
Hmmm, this might be subtly different from the term level `subZero a b = if a <= b then 0 else a - b`. As in, if `b` is bottom on the term level the equation will return bottom, but if `b` is bottom on the type level it will give an answer. I'm not 100% sure about this though, maybe type family resolution is strict in its arguments. But.. better safe than sorry, so lets remove this case?

Reply:
You are right, but unfortunately, we cannot drop the first case because of https://hackage.haskell.org/package/ghc-typelits-knownnat-0.7.10/docs/GHC-TypeLits-KnownNat.html (cf. FAQ).

Reply:
Oh wow, thanks. Could you link to that FAQ in the source?